lumail2.lua

```diff
@@ -1468,72 +1468,45 @@ end
 --
 function Message.delete ()
 
-  --
-  -- If we're in message-mode then we can get the current-message
-  -- directly.
-  --
-  -- If instead we're in index-mode then we'll need to select the message
-  -- under the cursor to proceed.
-  --
-  local mode = Config:get "global.mode"
-
-  if mode == "message" then
-
-    -- Get the current offset and the number of messages.
-    local cur = Config.get_with_default("index.current", 0)
-    local max = Config:get "index.max"
-
-    -- Get the message.
-    local msg = Global:current_message()
-
-    if not msg then
-      error_msg "Failed to find a message"
-      return
-    end
-
-    -- Delete the message.
-    msg:unlink()
-
-    -- Delete the message from the cache.
-    table.remove(global_msgs, cur + 1)
+  -- The Message we want to delete.
+  local msg = Message.at_point()
 
-    if cur <= (#global_msgs - 1) then
-      Global:select_message(global_msgs[cur + 1])
-      Config:set("index.current", cur)
-    end
+  -- If there is no message we have nothing to do.
+  if not msg then
+    error_msg "Failed to find message!"
     return
   end
 
-  if mode == "index" then
+  --
+  -- WARNING: index.current is 0 based!
+  --
+  local msg_index = Config.get_with_default("index.current", 0) + 1
 
-    -- Get the list of messages, and the current offset
-    -- that'll let us find the message.
-    local offset = Config.get_with_default("index.current", 0)
-    local msgs = get_messages()
-    if not msgs then
-      warning_msg "There are no messages!"
-      return
-    end
+  -- Move message to trash.
+  local trash = Config:get "global.trash-mail"
```

Comment:
In practice it doesn't matter, but I think it reads more cleanly.

As it was I applied it, unchanged. :)